server/http/src/server.rs

```diff
@@ -79,21 +81,29 @@ impl GraphQLServer {
     /// Handle incoming schema events.
     fn handle_schema_events(&mut self, stream: Receiver<SchemaEvent>) {
         let logger = self.logger.clone();
-        let schema = self.schema.clone();
+        let schemas = self.schemas.clone();
 
         tokio::spawn(stream.for_each(move |event| {
             info!(logger, "Received schema event");
 
             if let SchemaEvent::SchemaAdded(new_schema) = event {
-                let mut schema = schema.lock().unwrap();
                 let derived_schema = match api_schema(&new_schema.document) {
```

Comment:
I would like us to serve a subgraph with name `xyz` and ID `abc` via two URLs: `/xyz` and `/abc`. The simplest way to achieve that would probably be to insert the schema in the `schemas` map twice (once under the name, once under the subgraph ID) and then resolve both in the same way. Thoughts?